modules/opengl/include/modules/opengl/shader/shaderobject.h

```diff
@@ -132,6 +147,10 @@ class IVW_MODULE_OPENGL_API ShaderObject {
     bool hasShaderExtension(const std::string& extName) const;
     void clearShaderExtensions();
 
+    void addSegment(ShaderSegment segment);
+    void removeSegments(const std::string& key);
```

Comment:
Where does the key come from? 
Might want to return the key when adding a segment?